### ISSUE TITLE
Kernel: Panic when calling lspci on VirtualBox

### ISSUE THREAD

**Issue by supercomputer7** [SerenityOS/serenity#12446]
While I worked on fixing #12432 and fallout after #12411, I observed this:
```
0.000 [Kernel]: Loading kernel symbol table...
0.000 [Kernel]: CPU[0]: Supported features: nx pae pge rdrand rdseed sse tsc rdtscp constant_tsc nonstop_tsc sep syscall mmx sse2 sse3 ssse3 sse4.1 sse4.2 xsave avx fxsr lm hypervisor pat
0.000 [Kernel]: CPU[0]: Physical address bit width: 39
0.000 [Kernel]: CPU[0]: Virtual address bit width: 48
0.000 [#0 Kernel]: Initializing unhandled interrupt handlers
0.000 [Kernel]: CPU[0]: CPUID hypervisor signature 'KVMKVMKVM' (0x4b4d564b 0x564b4d56 0x4d), max leaf 0x40000001
0.000 [Kernel]: Kernel Commandline: serial_debug root=/dev/hda1
0.000 [Kernel]: MM: Multiboot mmap: address=0x00000000, length=654336, type=1
0.000 [Kernel]: MM: Got an unaligned physical_region from the bootloader; correcting length 654336 by 3072 bytes
0.000 [Kernel]: MM: Multiboot mmap: address=0x0009fc00, length=1024, type=2
0.000 [Kernel]: MM: Multiboot mmap: address=0x000f0000, length=65536, type=2
0.000 [Kernel]: MM: Multiboot mmap: address=0x00100000, length=1683947520, type=1
0.000 [Kernel]: MM: Multiboot mmap: address=0x646f0000, length=65536, type=3
0.000 [Kernel]: MM: Multiboot mmap: address=0xfec00000, length=4096, type=2
0.000 [Kernel]: MM: Multiboot mmap: address=0xfee00000, length=4096, type=2
0.000 [Kernel]: MM: Multiboot mmap: address=0xfffc0000, length=262144, type=2
0.000 [Kernel]: MM: Contiguous reserved range from P000000000009fc00, length is 394240
0.000 [Kernel]: MM: Contiguous reserved range from P00000000fec00000, length is 20971520
0.000 [Kernel]: MM: boot_pdpt @ P0000000000110000
0.000 [Kernel]: MM: boot_pd0 @ P0000000000111000
0.000 [Kernel]: MM: boot_pd_kernel @ P0000000000132000
0.000 [Kernel]: MM: Physical page entries: 0xc2601000 - 0xc2e01fff (size 0x00801000)
0.000 [Kernel]: MM: Low memory range @ P0000000000000000 - P00000000000fffff (size 0x100000)
0.000 [Kernel]: MM: Kernel range @ P0000000000200000 - P00000000025dcfff (size 0x23dd000)
0.000 [Kernel]: MM: Physical Pages range @ P00000000025de000 - P0000000002de3fff (size 0x806000)
0.000 [Kernel]: MM: Super physical region: P0000000000201000 - P0000000000600fff (size 0x400000)
0.000 [Kernel]:  * 4x PhysicalZone (1 MiB) @ 0000000000201000-0000000000500fff
0.000 [Kernel]: MM: User physical region: P0000000000101000 - P00000000001fefff (size 0xfe000)
0.000 [Kernel]: MM: User physical region: P0000000002de4000 - P00000000646effff (size 0x6190c000)
0.000 [Kernel]:  * 97x PhysicalZone (16 MiB) @ 0000000002de4000-0000000062de3fff
0.000 [Kernel]:  * 9x PhysicalZone (1 MiB) @ 0000000063de4000-00000000645e3fff
0.000 [Kernel]: Starting SerenityOS...
0.000 [#0 Kernel]: Early access to ACPI tables for interrupt setup
0.000 [#0 Kernel]: Can't enable SMP mode without IOAPIC mode being enabled
0.000 [Kernel]: Interrupts: Switch to IOAPIC mode
0.000 [#0 Kernel]: Interrupts: MADT @ P 0x646f0280
0.000 [Kernel]: PIC: Cascading mode, vectors 0x50-0x5f
0.000 [#0 Kernel]: Interrupts: Overriding INT 0x0 with GSI 2, for bus 0x0
0.000 [#0 Kernel]: Interrupts: Overriding INT 0x9 with GSI 9, for bus 0x0
0.000 [#0 Kernel]: IOAPIC found @ MADT entry 3, MMIO Registers @ P00000000fec00000
0.000 [Kernel]: IOAPIC ID: 0x0
0.000 [Kernel]: IOAPIC Version: 0x20, redirection entries: 24
0.000 [Kernel]: IOAPIC Arbitration ID 0xffffffff
0.000 [#0 Kernel]: Interrupts: Detected Dual i8259 - Disabled
0.000 [#0 Kernel]: Interrupts: Detected IOAPIC
0.000 [#0 Kernel]: MultiProcessor: Floating Pointer Structure @ P000000000009fff0
0.000 [#0 Kernel]: Features 0, IMCR? 0
0.000 [#0 Kernel]: MultiProcessor: Get PCI IOAPIC redirections
0.000 [#0 Kernel]: Trying to unregister unused handler (?)
0.000 [#0 Kernel]: Trying to unregister unused handler (?)
0.000 [#0 Kernel]: Trying to unregister unused handler (?)
0.000 [Kernel]: ACPI: Using RSDP @ P00000000000e0000
0.000 [Kernel]: ACPI: Main Description Table valid? true
0.000 [Kernel]: ACPI: Using XSDT, enumerating tables @ P00000000646f0040
0.000 [Kernel]: ACPI: XSDT revision 1, total length: 76
0.000 [Kernel]: ACPI: Initializing Fixed ACPI data
0.000 [Kernel]: ACPI: Fixed ACPI data, Revision 4, length: 244 bytes
0.000 [Kernel]: ACPI: DSDT P00000000646f0530
0.000 [Kernel]: RTC:  Year: 2022, month: 2, day: 11, hour: 21, minute: 34, second: 38
0.000 [Kernel]: HPET @ P00000000646f02e0
0.000 [Kernel]: HPET: Minimum clock tick - 4096
0.000 [Kernel]: HPET: Timers count - 4
0.000 [Kernel]: HPET: Main counter size: 64-bit
0.000 [Kernel]: HPET: Timer[0] comparator size: 64-bit, mode: 64-bit
0.000 [Kernel]: HPET: Timer[1] comparator size: 32-bit, mode: 32-bit
0.000 [Kernel]: HPET: Timer[2] comparator size: 32-bit, mode: 32-bit
0.000 [Kernel]: HPET: Timer[3] comparator size: 32-bit, mode: 32-bit
0.000 [Kernel]: HPET: frequency 14492753 Hz (14 MHz) resolution: 69 ns
0.000 [#0 Kernel]: Trying to unregister unused handler (?)
0.000 [#0 Kernel]: Trying to unregister unused handler (?)
0.000 [#0 Kernel]: HPET: Setting appropriate functions to timers.
0.000 [#0 Kernel]: Time: Scanning for periodic timers
0.000 [#0 Kernel]: Time: Scanning for non-periodic timers
0.000 [#0 Kernel]: Trying to unregister unused handler (?)
0.000 [Kernel]: APICTimer: Using HPET as calibration source
0.000 [Kernel]: KernelRng: Using RDSEED or RDRAND as entropy source
0.000 [Kernel]: APICTimer: Ticks per second: 264 (3.776527ms)
0.000 [Kernel]: APICTimer: Bus clock speed: 944.168160 MHz
0.000 [Kernel]: APICTimer: CPU clock speed: 3013.760090 MHz
0.000 [Kernel]: Time: Using APIC timer as system timer
0.184 [#0 colonel(0:0)]: Scheduler[0]: idle loop running
0.185 [#0 init_stage2(1:1)]: PCI: MCFG, length: 60, revision: 1
0.185 [init_stage2(1:1)]: PCI: New PCI domain @ P00000000dc000000, PCI buses (0-63)
0.206 [init_stage2(1:1)]: PCI [0000:00:02:00] PCI::HardwareID [80ee:beef]
0.206 [init_stage2(1:1)]: PCI [0000:00:03:00] PCI::HardwareID [8086:100e]
0.206 [init_stage2(1:1)]: PCI [0000:00:04:00] PCI::HardwareID [80ee:cafe]
0.206 [init_stage2(1:1)]: PCI [0000:00:05:00] PCI::HardwareID [8086:2415]
0.206 [init_stage2(1:1)]: PCI [0000:00:07:00] PCI::HardwareID [8086:7113]
0.206 [init_stage2(1:1)]: PCI [0000:00:1f:00] PCI::HardwareID [8086:27b9]
0.206 [init_stage2(1:1)]: PCI [0000:00:1f:01] PCI::HardwareID [8086:269e]
0.206 [init_stage2(1:1)]: PCI [0000:00:1f:04] PCI::HardwareID [106b:003f]
0.344 [#0 init_stage2(1:1)]: I8042: Dual channel controller
0.350 [init_stage2(1:1)]: PS2MouseDevice: Mouse wheel enabled!
0.368 [init_stage2(1:1)]: PS2MouseDevice: 5 buttons enabled!
0.382 [#0 init_stage2(1:1)]: Trying to unregister unused handler (?)
0.382 [#0 init_stage2(1:1)]: Trying to unregister unused handler (?)
0.384 [#0 init_stage2(1:1)]: Framebuffer Console: taking 3145728 bytes
0.384 [#0 init_stage2(1:1)]: BochsGraphicsAdapter: resolution set to 1024x768
0.384 [#0 init_stage2(1:1)]: Framebuffer Console: taking 3145728 bytes
0.384 [#0 init_stage2(1:1)]: Framebuffer 0: address=P00000000e0000000, pitch=4096, width=1024, height=768
0.384 [#0 init_stage2(1:1)]: Graphics adapter @ PCI [0000:00:02:00] is operating in VGA mode
0.393 [#0 init_stage2(1:1)]: VC 0: Resized to 128 x 96
0.393 [#0 init_stage2(1:1)]: VC 1: Resized to 128 x 96
0.395 [#0 init_stage2(1:1)]: VC 2: Resized to 128 x 96
0.395 [#0 init_stage2(1:1)]: VC 3: Resized to 128 x 96
0.395 [#0 init_stage2(1:1)]: VC 4: Resized to 128 x 96
0.398 [#0 init_stage2(1:1)]: VC 5: Resized to 128 x 96
0.398 [init_stage2(1:1)]: USBManagement: OHCI controller found at PCI [0000:00:1f:04] is not currently supported.
0.398 [#0 init_stage2(1:1)]: BIOSSysFSDirectory: SMBIOS 32bit Entry point @ P00000000000fff60
0.403 [#0 init_stage2(1:1)]: BIOSSysFSDirectory: Data table @ P00000000000e1000
0.403 [init_stage2(1:1)]: E1000: Found @ PCI [0000:00:03:00]
0.403 [init_stage2(1:1)]: E1000: port base: IO 0
0.403 [init_stage2(1:1)]: E1000: MMIO base: P00000000e4000000
0.403 [init_stage2(1:1)]: E1000: MMIO base size: 131072 bytes
0.403 [init_stage2(1:1)]: E1000: Interrupt line: 9
0.403 [init_stage2(1:1)]: E1000: Has EEPROM? true
0.403 [init_stage2(1:1)]: E1000: MAC address: 08:00:27:9a:cd:5c
0.403 [#0 init_stage2(1:1)]: Trying to unregister unused handler (?)
0.412 [#0 SyncTask(3:3)]: SyncTask is running
0.412 [#0 init_stage2(1:1)]: AC97: found audio controller at PCI [0000:00:05:00]
0.412 [#0 init_stage2(1:1)]: AC97 @ PCI [0000:00:05:00]: PCM front DAC rate set to 44100 Hz
0.412 [#0 init_stage2(1:1)]: AC97 @ PCI [0000:00:05:00]: channel PCMOut: resetting
0.412 [#0 init_stage2(1:1)]: Trying to unregister unused handler (?)
0.416 [#0 init_stage2(1:1)]: IDE controller @ PCI [0000:00:1f:01]: bus master base was set to IO d240
0.416 [#0 init_stage2(1:1)]: IDE controller @ PCI [0000:00:1f:01]: interrupt line was set to 0
0.416 [#0 init_stage2(1:1)]: IDE controller @ PCI [0000:00:1f:01]: ISA Compatibility mode controller, supports both channels switched to PCI native mode, supports bus mastering
0.416 [#0 init_stage2(1:1)]: IDE controller @ PCI [0000:00:1f:01]: primary channel DMA capable? 3
0.416 [#0 init_stage2(1:1)]: IDE controller @ PCI [0000:00:1f:01]: secondary channel DMA capable? 0
0.416 [#0 init_stage2(1:1)]: IDEChannel: Primary slave ATA device found: Name=VBOX HARDDISK, Capacity=676331520, Capabilities=0x0b00
0.416 [#0 init_stage2(1:1)]: Trying to unregister unused handler (?)
0.416 [#0 init_stage2(1:1)]: IDEChannel: Secondary master device doesn't appear to be ATA or ATAPI. Ignoring it.
0.416 [#0 init_stage2(1:1)]: IDEChannel: Secondary slave device doesn't appear to be ATA or ATAPI. Ignoring it.
0.416 [#0 init_stage2(1:1)]: Trying to unregister unused handler (?)
0.517 [init_stage2(1:1)]: VirtualFileSystem: mounted root from Ext2FS (device:100,0)
0.517 [init_stage2(1:1)]: Write-protected kernel symbols after init.
0.517 [init_stage2(1:1)]: Unmapped 1472 KiB of kernel text after init! :^)
0.529 [NetworkTask(6:6)]: NetworkTask: E1000NetworkAdapter network adapter found: hw=08:00:27:9a:cd:5c
0.529 [NetworkTask(6:6)]: NetworkTask: LoopbackAdapter network adapter found: hw=13:55:02:09:55:aa
0.577 SystemServer(5:5): Spawning mount -a to mount all filesystems.
0.585 mount(7:7): Mounting all filesystems...
0.588 mount(7:7): Skipping mounting root
0.588 mount(7:7): Mounting /bin (bind) on /bin
0.588 [#0 mount(7:7)]: mount bind: source fd 4 @ /bin
0.588 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:13761 at inode 1:13761
0.592 mount(7:7): Mounting /etc (bind) on /etc
0.592 [#0 mount(7:7)]: mount bind: source fd 5 @ /etc
0.592 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:20641 at inode 1:20641
0.592 mount(7:7): Mounting /home (bind) on /home
0.592 [#0 mount(7:7)]: mount bind: source fd 6 @ /home
0.592 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:27521 at inode 1:27521
0.596 mount(7:7): Mounting /root (bind) on /root
0.596 [#0 mount(7:7)]: mount bind: source fd 7 @ /root
0.596 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:27522 at inode 1:27522
0.596 mount(7:7): Mounting /var (bind) on /var
0.596 [#0 mount(7:7)]: mount bind: source fd 8 @ /var
0.600 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:14882 at inode 1:14882
0.600 mount(7:7): Mounting /www (bind) on /www
0.600 [#0 mount(7:7)]: mount bind: source fd 9 @ /www
0.600 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:13762 at inode 1:13762
0.600 mount(7:7): Mounting /usr/Tests (bind) on /usr/Tests
0.600 [#0 mount(7:7)]: mount bind: source fd 10 @ /usr/Tests
0.604 [#0 mount(7:7)]: VirtualFileSystem: Bind-mounting inode 1:8506 at inode 1:8506
0.604 mount(7:7): Mounting none (tmp) on /tmp
0.604 [#0 mount(7:7)]: mount tmp @ /tmp
0.604 [#0 mount(7:7)]: VirtualFileSystem: Mounting TmpFS at inode 1:14880 with flags 5
0.611 [#0 SystemServer(5:5)]: mount proc @ /proc
0.611 [#0 SystemServer(5:5)]: VirtualFileSystem: Mounting ProcFS at inode 1:14878 with flags 4
0.611 [#0 SystemServer(5:5)]: mount sys @ /sys
0.611 [#0 SystemServer(5:5)]: VirtualFileSystem: Mounting SysFS at inode 1:14886 with flags 0
0.611 [#0 SystemServer(5:5)]: mount dev @ /dev
0.611 [#0 SystemServer(5:5)]: VirtualFileSystem: Mounting DevTmpFS at inode 1:14885 with flags 0
0.615 [#0 SystemServer(5:5)]: mount devpts @ /dev/pts
0.615 [#0 SystemServer(5:5)]: VirtualFileSystem: Mounting DevPtsFS at inode 5:26 with flags 0
0.629 SystemServer(5:5): Creating /tmp/coredump directory
0.633 SystemServer(5:5): Read system_mode: graphical
0.633 SystemServer(5:5): System in graphical mode
0.641 SystemServer(5:5): Activating 21 services...
0.697 SystemServer(5:5): Service KeyboardPreferenceLoader has exited with exit code 0
0.723 LookupServer(17:17): Using network config file at /etc/LookupServer.ini
0.745 AudioServer(18:18): Started thread "AudioServer[mixer]", tid = 20
0.753 [#0 keymap(19:19)]: New Character map 'en-us' passed in by client.
0.782 SpiceAgent(13:13): Exiting with runtime error: unveil: No such file or directory (errno=2)
0.786 SystemServer(5:5): Service SpiceAgent has exited with exit code 256
0.808 WindowServer(16:16): update_virtual_rect for screen #0: [0,0 1024x768]
0.808 [#0 WindowServer(16:16)]: BochsGraphicsAdapter: resolution set to 1024x768
0.808 [#0 WindowServer(16:16)]: Framebuffer Console: taking 3145728 bytes
0.823 WindowServer(16:16): update_virtual_rect for screen #0: [0,0 1024x768]
0.823 WindowServer(16:16): update_virtual_rect for screen #0: [0,0 1024x768]
0.932 WindowServer(16:16): Entering WindowServer main loop
0.976 LoginServer(21:21): login with sid=21
0.980 SystemServer(21:21): Activating 9 services...
1.081 ResourceGraph.Applet(30:30): Create applet: 0 with spec 'CPUGraph,#00bb00'
1.163 ResourceGraph.Applet(30:30): Create applet: 1 with spec 'MemoryGraph,#00bbbb'
1.197 utmpupdate(32:32): Updating utmp from UID=100 GID=100 EGID=5 PID=31
1.420 [#0 SyncTask(3:3)]: Ext2FS: Flushed 8 blocks to disk
4.443 [#0 colonel(0:0)]: BMIDEChannel: print_ide_status: DRQ=false BSY=false, DRDY=true, DSC=false, DF=false, CORR=false, IDX=false, ERR=true
4.443 [#0 colonel(0:0)]: BMIDEChannel: Error 0x10!
[colonel(0:0)]: ASSERTION FAILED: m_lock.is_locked()
[colonel(0:0)]: ./Kernel/Storage/ATA/IDEChannel.cpp:160 in void Kernel::IDEChannel::try_disambiguate_error()
[colonel(0:0)]: KERNEL PANIC! :^(
[colonel(0:0)]: Aborted
[colonel(0:0)]: at ./Kernel/Arch/x86/common/CPU.cpp:35 in void abort()
[colonel(0:0)]: Kernel + 0x00c200c1  Kernel::__panic(char const*, unsigned int, char const*) +0xf1
[colonel(0:0)]: Kernel + 0x00f8cf79  abort.localalias +0x245
[colonel(0:0)]: Kernel + 0x00f8cd34  abort.localalias +0x0
[colonel(0:0)]: Kernel + 0x006aec6f  .L5371 +0x3f
[colonel(0:0)]: Kernel + 0x006a11b1  Kernel::BMIDEChannel::handle_irq(Kernel::RegisterState const&) [clone .localalias] +0x9a1
[colonel(0:0)]: Kernel + 0x00f92da8  handle_interrupt +0x328
[colonel(0:0)]: Kernel + 0x00f8b02b  interrupt_common_asm_entry +0x2b
[colonel(0:0)]: Kernel + 0x00f956d0  exit_kernel_thread +0x0
```

**Comment by diversys** [comment]
Can't reproduce.

**Comment by supercomputer7** [comment]
This has been fixed already in [95238bdb1650740dfb530091fd5cd0d195f17e68](https://github.com/SerenityOS/serenity/commit/95238bdb1650740dfb530091fd5cd0d195f17e68).
